### PR TITLE
Added support for Steelseries Heroes of the Storm (Sensei Raw) Gaming Mouse

### PR DESCRIPTION
**Pull request by frags51**
Change sensitivity, light effects and brightness configure mouse buttons.

### CODE REVIEW

**Comment by flozz** [comment]
Hi, I merge your PR on a branch based on `road-to-3-0-0-hidapi` and I will adapt your work to the new architecture of the software. I will then need you to check if it work or not :)

**Comment by frags51** [comment]
Allright. Whatever you feel correct!

File: rivalcfg/helpers.py [hunk 1]
```diff
@@ -118,6 +140,41 @@ def color_string_to_rgb(color_string):
...
+    Returns a list of strings
+    """
+    kstring=kstring.split()
+    outlist = []
+ 
+    #the list of commands to be outputted
+    if not len(kstring) == 8:
+        raise ValueError("Invalid length of argument to Set Button Commands!")
+    for x in range(0, 8):
+        kstring[0] = kstring[0][1:] #First char of first string is '=' for some reason
```
[the remark this answers is not in the document]
Comment (frags51) [reply rivalcfg/helpers.py]: Oh! Actually I don't have much experience in Python / optparse, that's why I was confused. I generally code in C/Java

**Comment by flozz** [comment]
Ok, I adapted you work for the new branch. Can you test if it works?

https://github.com/flozz/rivalcfg/tree/rivalcfg-3-0-0-steelseries-hos

I tried in DRY mode and it seems to output a valid command:

```
$ RIVALCFG_DRY=1 RIVALCFG_PROFILE=1038:1390 python -m rivalcfg --set-btns="a b c d e f g h"
[DEBUG] Rivalcfg 2.6.0
[DEBUG] Python version: 2.7.13
[DEBUG] OS: Linux
[DEBUG] Linux distribution: Ubuntu 17.04 zesty
[DEBUG] Dry run enabled
[DEBUG] Forced profile: 1038:1390
[DEBUG] Targeted device: 1038:1390
[DEBUG] Selected mouse: <Mouse SteelSeries Heroes of the Storm (Sensei Raw) (1038:1390:00)>
[DEBUG] Mouse._device_write [wValue]: 02 00
[DEBUG] Mouse._device_write   [data]: 31 00 10 04 00 10 05 00 10 06 00 10 07 00 10 08 00 10 09 00 10 0A 00 10 0B 00
[DEBUG] Mouse._device_write [wValue]: 02 00
[DEBUG] Mouse._device_write   [data]: 09 00
```

File: rivalcfg/helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (flozz) [reply rivalcfg/helpers.py]: No problem, but now you know why there was a "=" in your string :)

**Comment by frags51** [comment]
I am currently having some issues in getting this to work. My computer hangs when I try pip install hidapi, or pip install rivalcfg
This is the message I see:
Collecting hidapi>=0.7.99.post20 (from rivalcfg==2.6.0)
  Using cached hidapi-0.7.99.post21.tar.gz

It doesn't proceed further.

**Comment by flozz** [comment]
It can be a bit long to process as it have to compile some C code (hidapi is a Python binding for a C library).

NOTE: You will need gcc and Python headers to compile/install it. On Debian / Ubuntu you can install the `build-essential` and `python-dev` package

**Comment by frags51** [comment]
I already had build-essential and python-dev. But after I ran installed these: `sudo apt-get install cython libhidapi-dev libudev-dev libusb-1.0-0 libusb-1.0-0-dev` 

Anyways, now it works, but it is showing: 

> W: The following mouse was found but rivalcfg was not able to open it:
>  * mouse: SteelSeries Heroes of the Storm (Sensei Raw) (1038:1390)
>  * error: open failed

**Comment by flozz** [comment]
This error is caused because the `rivalcfg/data/99-steelseries-rival.rules`was not copied in `/etc/udev/rules.d/` (in this branch there is new rules inside this file).

So:
1. Copy/update this ile
2. run the `sudo udevadm trigger` command to update rules

It should work then :)

**Comment by frags51** [comment]
Did that multiple times, still not working. Dry run is going fine.

**Comment by flozz** [comment]
Strange it should work :(

Is the mouse attached to a VM or something like that?

**Comment by frags51** [comment]
No, I am not using any VM. The mouse is showing up in xinput/dmesg

**Comment by flozz** [comment]
With those two lines in the udev rule files it should have worked :(

```
SUBSYSTEM=="hidraw", ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1390", MODE="0666"
SUBSYSTEM=="usb", ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1390", MODE="0666"
```

I will try to reproduce your issue tomorrow

**Comment by flozz** [comment]
Hello,

I am not able to reproduce with my Rival 100 and 300 without removing the `SUBSYSTEM=="usb" [...]` lines in the udev rules file.

* Can you tell me what is your linux distro and its version?
* Does it work if you run the command as root?

**Comment by flozz** [comment]
Hi, any additional info ? :)

**Comment by frags51** [comment]
Hi, I was sick hence I could not do anything. The software began working now. I don't know what the error was, probably just a restart was needed?

**Comment by flozz** [comment]
Hi 

The `sudo udevadm trigger` command should reload the udev rules, but maybe it not worked. Anyway, I will merge my branch on `road-to-3-0-0-hidapi` and merge it back to master. This will allow everyone to work on the new version of the software instead of the old one :)

Thank you for your work ad your feedbacks! :D

**Comment by flozz** [comment]
Hello, it is on master :), more info here → https://github.com/flozz/rivalcfg/issues/42

**Pull request closed without merging**